homeassistant/components/vallox/__init__.py

```diff
@@ -293,3 +318,38 @@ async def async_handle(self, call: ServiceCall) -> None:
         # be observed by all parties involved.
         if result:
             await self._coordinator.async_request_refresh()
+
+
+class ValloxEntity(CoordinatorEntity):
+    """Representation of a Vallox entity."""
+
+    coordinator: ValloxDataUpdateCoordinator
+    _device_config_url: str | None
+
+    def __init__(self, name: str, coordinator: ValloxDataUpdateCoordinator) -> None:
+        """Initialize a Vallox entity."""
+        super().__init__(coordinator)
+
+        if self.coordinator.config_entry is not None:
+            self._device_config_url = (
+                f"http://{self.coordinator.config_entry.data[CONF_HOST]}"
+            )
+        else:
+            self._device_config_url = None
+
+        self._device_uuid = self.coordinator.data.get_uuid()
+        self._device_model = self.coordinator.data.get_model()
+        self._device_name = name
+        self._device_sw_version = self.coordinator.data.get_sw_version()
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device information of the entity."""
+        return DeviceInfo(
+            configuration_url=self._device_config_url,
+            identifiers={(DOMAIN, str(self._device_uuid))},
```

Comment:
Please use cast() or adhere to https://www.python.org/dev/peps/pep-0484/, if you only use `str()` for typing purposes.

Reply:
`self._device_uuid` is type `UUID`, so str() is needed here, right?